src/project-generator.ts

```diff
@@ -51,266 +40,80 @@ export class ProjectGenerator {
-        const maxRetries = 20;
-
-        while (retries < maxRetries && !success) {
-            try {
-                const response = await axios.get(`${this.repositoryPath}/contents`, this.requestConfig);
-                responseStatus = response.status;
-                success = true;
-                return responseStatus;
-            } catch (error) {
-                if (axios.isAxiosError(error)) {
-                    console.log(`Check #${retries + 1} if Repository is generated failed. Retrying.`);
-                    responseStatus = error.response?.status as number;
-                } else {
-                    throw error;
-                }
-            }
-            retries++;
-        }
-        return responseStatus;
-    }
-
-    private async enableWorkflows(isEnabled: boolean): Promise<boolean> {
-        try {
-            await axios.put(
-                `${this.repositoryPath}/actions/permissions`,
-                {
-                    enabled: isEnabled,
-                },
-                this.requestConfig
-            );
-            return true;
-        } catch (error) {
-            console.log(error);
-            return false;
-        }
-    }
-
-    private async setDefaultWorkflowPermissionToWrite(): Promise<boolean> {
-        try {
-            await axios.put(
-                `${this.repositoryPath}/actions/permissions/workflow`,
-                {
-                    default_workflow_permissions: 'write',
-                },
-                this.requestConfig
-                );
-            return true;
-        } catch (error) {
-            console.log(error);
-            return false;
-        }
-    }
-
-    private async createBlob(fileContent: string): Promise<string> {
+    /**
+     * @param {string} codeSnippet Base64 encoded playground code snippet.
+     * @param {string} appName Name of the VehicleApp.
+     * @param {VspecUriObject} VspecUriObject Containing Repo and Commit hash.
+     * @throws {ProjectGeneratorError}
+     */
+    public async runWithUri(codeSnippet: string, appName: string, vspecUriObject: VspecUriObject): Promise<number> {
         try {
-            const response = await axios.post(
-                `${this.repositoryPath}/git/blobs`,
-                {
-                    content: fileContent,
-                    encoding: CONTENT_ENCODINGS.base64,
-                },
-                this.requestConfig
-            );
-            const blobSha = response.data.sha;
-            return blobSha;
-        } catch (error) {
-            if (axios.isAxiosError(error)) {
-                throw new ProjectGeneratorError(error);
-            } else {
-                throw error;
-            }
-        }
-    }
+            // Assumption for now is, that all individual vspecs are a fork of COVESA following this path
+            const vspecUriString = `${vspecUriObject.repo}/tree/${vspecUriObject.commit}/spec`;
```

Comment:
OK, this an issue! DA is pointing to a vspec (yaml) tree here, which we don't support well with a single URL. I need to think about this :-(

Reply:
Adapted to not export runWithUri for now, as discussed, we will proceed iteratively and first only support getting the vspec payload always